crates/mongodb-agent-common/src/query/arguments/json_to_bson.rs

```diff
@@ -0,0 +1,424 @@
+        JsonToBsonError::ConversionErrorWithContext(Type::Scalar(expected_type), value, err.into())
+    })
+}
+
+fn from_string(expected_type: BsonScalarType, value: Value) -> Result<String> {
+    match value {
+        Value::String(s) => Ok(s),
+        _ => Err(JsonToBsonError::IncompatibleBackingType {
+            expected_type: Type::Scalar(expected_type),
+            expected_backing_type: "String",
+            value,
+        }),
+    }
+}
+
+fn incompatible_scalar_type<T>(expected_type: BsonScalarType, value: Value) -> Result<T> {
+    Err(JsonToBsonError::IncompatibleType(
+        Type::Scalar(expected_type),
+        value,
+    ))
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::BTreeMap, str::FromStr};
+
+    use configuration::schema::{ObjectField, ObjectType, Type};
+    use mongodb::bson::{self, datetime::DateTimeBuilder, Bson};
+    use mongodb_support::BsonScalarType;
+    use pretty_assertions::assert_eq;
+    use serde_json::json;
+
+    use super::json_to_bson;
+
+    #[test]
+    fn deserializes_specialized_scalar_types() -> anyhow::Result<()> {
+        let object_type_name = "scalar_test".to_owned();
+        let object_type = ObjectType {
+            fields: BTreeMap::from([
+                ObjectField::new("double", Type::Scalar(BsonScalarType::Double)),
+                ObjectField::new("int", Type::Scalar(BsonScalarType::Int)),
+                ObjectField::new("long", Type::Scalar(BsonScalarType::Long)),
+                ObjectField::new("decimal", Type::Scalar(BsonScalarType::Decimal)),
+                ObjectField::new("string", Type::Scalar(BsonScalarType::String)),
+                ObjectField::new("date", Type::Scalar(BsonScalarType::Date)),
+                ObjectField::new("timestamp", Type::Scalar(BsonScalarType::Timestamp)),
+                ObjectField::new("binData", Type::Scalar(BsonScalarType::BinData)),
+                ObjectField::new("objectId", Type::Scalar(BsonScalarType::ObjectId)),
+                ObjectField::new("bool", Type::Scalar(BsonScalarType::Bool)),
+                ObjectField::new("null", Type::Scalar(BsonScalarType::Null)),
+                ObjectField::new("undefined", Type::Scalar(BsonScalarType::Undefined)),
+                ObjectField::new("regex", Type::Scalar(BsonScalarType::Regex)),
+                ObjectField::new("javascript", Type::Scalar(BsonScalarType::Javascript)),
+                ObjectField::new(
+                    "javascriptWithScope",
+                    Type::Scalar(BsonScalarType::JavascriptWithScope),
+                ),
+                ObjectField::new("minKey", Type::Scalar(BsonScalarType::MinKey)),
+                ObjectField::new("maxKey", Type::Scalar(BsonScalarType::MaxKey)),
+                ObjectField::new("symbol", Type::Scalar(BsonScalarType::Symbol)),
+            ]),
+            description: Default::default(),
+        };
+
+        let input = json!({
+            "double": 3.14159,
```

Comment:
Clippy really doesn't like you trying to impersonate pi. I found that with one of my test cases too 😂 

https://github.com/hasura/ndc-mongodb/actions/runs/8446549353/job/23135506300?pr=16#step:6:163

Reply:
That's hilarious. And something I didn't realize is that the CI setup is running lints on test code, while it looks like when I run `cargo clippy` locally it doesn't check anything annotated with `#[cfg(test)]`